src/main/java/com/iota/iri/service/milestone/impl/MilestoneSolidifierImpl.java

```diff
@@ -197,6 +197,7 @@ private void processSolidifyQueue() throws Exception {
             switch(validity) {
                 case VALID:
                     milestoneCandidate.isMilestone(tangle, snapshotProvider.getInitialSnapshot(), true);
+                    registerNewMilestone(getLatestMilestoneIndex(), milestoneIndex, milestoneHash);
```

Comment:
I am actually not sure why it is a bug to wait until the milestone is solid before registering...
Care to elaborate?

Reply:
That was not the bug as it happens. The issue was in the propagation stalling out, but since I had made this change prior to fixing that I was under the false assumption that this part of it was necessary. As it turns out it does not look to be the case, so I've reverted it to only register milestones when being added to the seen milestone queue. Same as before, but a little cleaner with the check in the `registerNewMilestone` method.